src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -1345,6 +1345,16 @@ convert_aclright_to_string(int aclright)
 			return "TEMPORARY";
 		case ACL_CONNECT:
 			return "CONNECT";
+#if PG_VERSION_NUM >= PG_VERSION_15
+		case ACL_SET:
+			return "SET";
+		case ACL_ALTER_SYSTEM:
+			return "ALTER SYSTEM";
+#endif
+#if PG_VERSION_NUM >= PG_VERSION_16
```

Comment:
I think adding one test for:

> ACL_MAINTAIN
https://github.com/postgres/postgres/commit/60684dd834a222fefedd49b19d1f0a6189c1632e

sounds interesting. We create a distributed table, and for a non-owning user, we give MAINTAIN right.

Then, we try `VACUUM, ANALYZE, REINDEX,  CLUSTER, and LOCK TABLE.` commands with that user. Lets make sure there is no gap in the processing of this from Citus side.

I'm fine to merge this now, but add a test to PG16 branch.

Reply:
It looks like MAINTAIN will be reverted in PG16BETA3 https://github.com/postgres/postgres/commit/957445996fda2d6939a8748f2a19c10b15941c5e

I will remove this commit from this PR.